d2/src/main/java/com/linkedin/d2/balancer/simple/SimpleLoadBalancer.java

```diff
@@ -821,9 +829,43 @@ public void getLoadBalancedServiceProperties(String serviceName, boolean waitFor
     }
   }
 
+  public void processGetLoadBalancedServicePropertiesTimeoutError(
+          Throwable e, String serviceName, Callback<ServiceProperties> servicePropertiesCallback)
+  {
+    ServiceProperties properties = getServicePropertyFromCache(serviceName, servicePropertiesCallback);
+    if (properties != null)
+    {
+      _log.info("get service properties timeout, so read from the cache successfully, service is {}", serviceName);
+      servicePropertiesCallback.onSuccess(properties);
+    }
+    else
+    {
+      _log.error("get service properties timeout, and there is no value in cache, service is {} ", serviceName);
+      servicePropertiesCallback.onError(new ServiceUnavailableException(serviceName, "PEGA_1011. " + e.getMessage(),
+              e));
+    }
+  }
+
+  public ServiceProperties getServicePropertyFromCache(String serviceName,
+                                                       Callback<ServiceProperties> servicePropertiesCallback)
+  {
+    LoadBalancerStateItem<ServiceProperties> serviceItem = _state.getServiceProperties(serviceName);
+
+    if (serviceItem == null || serviceItem.getProperty() == null)
+    {
+      warn(_log, "unable to find service: ", serviceName);
+      _serviceNotFoundStats.inc();
+      die(servicePropertiesCallback, serviceName, "PEGA_1012. no service properties in lb state");
+      return null;
+    }
+
+    debug(_log, "got service: ", serviceItem);
+    return serviceItem.getProperty();
+  }
+
   @Override
   public void getLoadBalancedClusterAndUriProperties(String clusterName,
-      Callback<Pair<ClusterProperties, UriProperties>> callback)
+                                                     Callback<Pair<ClusterProperties, UriProperties>> callback)
```

Comment:
Bad alignment :)

Reply:
Actually there are lot's of this format code which is not written by me. But let me fix this.